cli/src/commands.rs

```diff
@@ -33,6 +33,10 @@ pub(crate) enum Command {
         #[arg(long)]
         address: Option<String>,
 
+        /// BIP353 address that contains a BOLT12 offer. If specified, amount_sat must also be set.
+        #[arg(long)]
+        pay_code: Option<String>,
```

Comment:
Would rename this to something more explicit like `bip353_address`

Reply:
Maybe we even don't need this. A bip353 address resolves to a bip21 URI, so functionally, it shouldn't be different from the `address` field.

Reply:
Bip353 addresses can resolve to one of multiple alternatives; some don't specify an amount (like BOLT12), but others do. Since we cannot know without resolving the address, we should not impose an amount here. 

As long as the `parse` method can find out precisely what the bip353 address resolves to, it's okay. SDK users can use it to figure out if an amount is needed or not.